sirepo/sim_api/jupyterhublogin.py

```diff
@@ -41,47 +41,19 @@ async def api_checkAuthJupyterHub(self):
     @sirepo.quest.Spec("require_user", sim_type=f"SimType const={_SIM_TYPE}")
     async def api_redirectJupyterHub(self):
         self.parse_params(type=_SIM_TYPE)
-        u = _unchecked_jupyterhub_user_name(self)
-        if u:
-            return self.reply_redirect("jupyterHub")
-        if not _cfg.rs_jupyter_migrate:
-            if not u:
-                create_user(self)
-            return self.reply_redirect("jupyterHub")
-        return self.reply_ok()
+        if not _unchecked_jupyterhub_user_name(self):
+            create_user(self)
+        return self.reply_redirect("jupyterHub")
 
 
 def cfg():
     return _init()
 
 
-def create_user(qcall, github_handle=None, check_dir=False):
-    """Create a Jupyter user and possibly migrate their data from old jupyter.
-
-    Terminology:
-      migration user: A user with data at the old jupyter
-      jupyter user: The user of new jupyter
-
-    A few interesting cases to keep in mind:
-      1. If a user is migrating (has a github handle) we should never modify
-         the handle and if they are able to migrate then their username should be
-         their github handle (downcased).
-      2. User signs into sirepo under one@any.com. They migrate their data using
-         GitHub handle y. They sign into sirepo under two@any.com. They choose
-         to migrate GitHub handle y again. We should let them know that they
-         have already migrated.
-      3. one@any.com signs up for jupyter and does not migrate data. They are
-         given the username one. two@any.com signs up for jupyter and they
-         migrate their data, but they have no data to migrate. They have the
-         github handle one and no previous data. They should be alerted that
-         they can't migrate that GitHub handle.
-      4. A new user signs in with foo@any.com and they do not select to
```

Comment:
still valid and maybe the other too

Reply:
Can you elaborate?

Reply:
I think this is the default case when they had a github handle with a jupyter address. There will be a collision. Maybe rewording the 1-4 text to describe the case for normal collisions.

Reply:
[d3a035f](https://github.com/radiasoft/sirepo/pull/7027/commits/d3a035fba7ef0e4619416db1861c69dc51efe402)